tests/test_insert_g8_framework_results.py

```diff
@@ -0,0 +1,191 @@
+        "yourself without the use of third parties (subcontractors)"
+    ],
+    "tradingStatus": "limited company",
+    "terrorism": False,
+    "conflictOfInterest": False,
+    "proofOfClaims": True,
+    "status": "complete",
+    "witheldSupportingDocuments": False,
+    "registeredAddressTown": "Cardigan",
+    "currentRegisteredCountry": "UK",
+    "misleadingInformation": False,
+    "equalityAndDiversity": True,
+    "nameOfOrganisation": "Cardigan Cardigans Ltd",
+    "fraudAndTheft": False,
+    "firstRegistered": "2001",
+    "employersInsurance": "Yes",
+    "unspentTaxConvictions": False,
+    "establishedInTheUK": True,
+    "servicesHaveOrSupport": True,
+    "influencedContractingAuthority": False,
+    "readUnderstoodGuidance": True,
+    "environmentalSocialLabourLaw": False,
+    "termsOfParticipation": True,
+    "accuratelyDescribed": True,
+    "organisedCrime": False,
+    "MI": True,
+    "cyberEssentialsPlus": False,
+    "registeredAddressPostcode": "CD3 4GH",
+    "tradingNames": "Cardies'r'Us"
+}
+
+
+def test_insert_result_calls_with_correct_arguments(mock_data_client):
+    assert insert_result(mock_data_client, 123456, True, 'user') == '  Result set OK: 123456'
+    mock_data_client.set_framework_result.assert_called_with(123456, 'g-cloud-8', True, 'user')
+
+
+def test_insert_result_returns_error_message_if_update_fails(mock_data_client):
+    mock_data_client.set_framework_result.side_effect = HTTPError()
+    assert insert_result(mock_data_client, 123456, True, 'user') == \
+        '  Error inserting result for 123456 (True): Request failed (status: 503)'
+
+
+def test_get_submitted_drafts_calls_with_correct_arguments(mock_data_client):
+    mock_data_client.find_draft_services.return_value = {'services': []}
+    get_submitted_drafts(mock_data_client, 12345)
+    mock_data_client.find_draft_services.assert_called_with(12345, framework='g-cloud-8')
+
+
+def test_get_submitted_drafts_returns_submitted_services_only(mock_data_client):
+    mock_data_client.find_draft_services.return_value = {'services': [
+                                                        {"id": 1, "status": "not-submitted"},
+                                                        {"id": 2, "status": "submitted"}
+        ]
+    }
+    result = get_submitted_drafts(mock_data_client, 12345)
+    assert (result == [{"id": 2, "status": "submitted"}])
+
+
+def test_check_declaration_fails_incomplete_declaration():
+    declaration = {"status": "started"}
+    assert check_declaration_answers(declaration) == 'Fail'
+
+
+def test_check_declaration_answers_passes_good_declaration():
+    assert check_declaration_answers(VALID_COMPLETE_G8_DECLARATION) == 'Pass'
+
+
+def test_check_declaration_answers_fails_bad_declaration_true_is_false():
+    declaration_content = mock.Mock()
+    question_numbers = mock.PropertyMock(side_effect=[1, 17, 21, -1])
+    type(declaration_content.get_question.return_value).number = question_numbers
```

Comment:
Good question!  This comes from the (copied) DOS version of the tests which relied on question numbers rather than keys.  I thought I'd fixed them all to the new key-based version but must have missed this one.

It passes green though :)